ADR/0027-tekton-results-naming-convention.md

```diff
@@ -0,0 +1,152 @@
+
+The standards presented in this ADR supersede those in [ADR 13. AppStudio Test Stream - API contracts](0013-integration-service-api-contracts.html). All other standards presented in the previous ADR hold unless _this_ ADR is superseded by an additional ADR that deprecates those standards. These other non-deprecated standards presented include [Detailed Conftest Output JSON](0013-integration-service-api-contracts.html#detailed-conftest-output-json), [Information injection](0013-integration-service-api-contracts.html#information-injection), [Information format](0013-integration-service-api-contracts.html#information-format), and [Image references](0013-integration-service-api-contracts.html#image-references).
+
+### Results for Test-like Tasks
+Test-like tasks are those whose results can be immediately classified as successful or failed. If these tests are taken into account in the final suitability of an artifact for promotion or release, then all pass/fail determination would be deferred to the task run.
+
+The output of each Tekton task will be provided in a minimized [Tekton result](https://tekton.dev/docs/pipelines/tasks/#emitting-results) in JSON format listing all test failures. The name of the result will be **TEST_OUTPUT**.
+
+#### Tekton Result Format for `TEST_OUTPUT`
+
+The Test output of the Tekton result **TEST_OUTPUT** will be a JSON object that includes the context about the test along with the list of check names for all failed checks
+
+The output will provide the following information about the overall test result:
+- **result** - The outcome of the testing task, can be `SUCCESS`, `FAILURE`, `WARNING`, `SKIPPED` or `ERROR`
+- **namespace** - The rego namespace for the test policy. If not set, it will be assumed to have the value `default`
+- **timestamp** - An UNIX epoch timestamp of the test completion time
+- **successes** - The number of successful checks in the form of an integer
+- **note** - A short note provided by test workstream to provide additional information about the test
+- **failures** - The number of failed checks in the form of an integer
+- **warnings** - The number of warning checks in the form of an integer
+
+
+#### `TEST_OUTPUT` Schema Validation
+The output of the Tekton result **TEST_OUTPUT** will be validated using the [jsonschema validator package](https://github.com/santhosh-tekuri/jsonschema).
+The schema is configured as follows:
+
+```
+{
+  "$schema": "http://json-schema.org/draft/2020-12/schema#",
+  "type": "object",
+  "properties": {
+    "result": {
+      "type": "string",
+      "enum": ["SUCCESS", "FAILURE", "WARNING", "SKIPPED", "ERROR"]
+    },
+    "namespace": {
+      "type": "string"
+    },
+    "timestamp": {
+      "type": "string",
+      "pattern": "^[0-9]{10}$"
```

Comment:
If this is a string why not prescribe it to be in RFC3339 format?

Reply:
@sonam1412 @dirgim, would this be a change that we can propose in this ADR -- to change the timestamp validation?

Reply:
I think we could make the change - we'd need to implement the change on all sides though since we're all using a string currently.

Reply:
> I think we could make the change to use the int for the timestamp

What would be the proposed pattern to follow? `YYYY-MM-DDTHH:MM:SSZ`?

Reply:
My 2c, either we change the type to a number, or we change to a standard format (RFC3339).

Reply:
I think RFC3339 format is more readable.

Reply:
Thinking about this some more, I also agree that we can switch to RFC3339 (So e.g. `2019-10-12T07:20:50.52Z`) and keep the string type for this field.

Reply:
This has been modified in the most recent commit.